Add pretty repr to `Effect` base class to avoid cmds spam

teutoburg opened the pull request:
Should be super useful for debugging as well as notebooks.

oczoske reviewed (approved):
Nice. The problem in my comment originates elsewhere, and of course we could always write better `__str__`s.

teutoburg commented:
> Should I see an effect when I look at a `SpectralTraceList`?
> 
> ```
> In [27]: metis['lms_spectral_traces']
> Out[27]: 
> MetisLMSSpectralTraceList(**{'filename': '!OBS.trace_file', 'description': 'list of spectral order trace geometry on the focal plane', 'history': ['Opened handle to FITS file TRACE_LMS.fits'], 'name': 'lms_spectral_traces', 'pixel_scale': 0.0082, 
> ...
> In [28]: print(metis['lms_spectral_traces'])
> SpectralTraceList: "lms_spectral_traces" : 28 traces
> ```
> 
> It works on `metis['psf']`, though.
> 
> Maybe because `SpectralTraceList` defines its own `__repr__`?

Well spotted. Although the new method uses `str(self)`, for some reason I don't fully understand, the definition of the `SpectralTraceList.__repr__()` (the base class of `MetisLMSSpectralTraceList` for those reading along) was confusing it. Which was redundant anyway, because it just redefined _its_ base class implementation of the same method, which looked exactly identical. I removed the redundant one and now the printout looks as expected. Also fixed the corresponding `__str__`, which had the class name hardcoded, so the derived subclass would show it wrong.
